packages/js/client/src/plugin/PluginWrapper.ts

```diff
@@ -78,9 +78,7 @@ export class PluginWrapper extends Wrapper {
       await this._sanitizeAndLoadEnv(client, module);
 
       let jsInput: Record<string, unknown>;
-
-      // If the input is a msgpack buffer, deserialize it
-      if (input instanceof ArrayBuffer) {
+      if (input instanceof ArrayBuffer || input instanceof Uint8Array) {
```

Comment:
What about `if(input instanceof ArrayBufferView)`
Since:
```typescript
  type TypedArray =
        | Uint8Array
        | Uint8ClampedArray
        | Uint16Array
        | Uint32Array
        | Int8Array
        | Int16Array
        | Int32Array
        | BigUint64Array
        | BigInt64Array
        | Float32Array
        | Float64Array;
    type ArrayBufferView = TypedArray | DataView;
```